lib/engines/content_block_manager/app/assets/javascripts/content_block_manager/modules/copy-embed-code.js

```diff
@@ -0,0 +1,69 @@
+'use strict'
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {}
+;(function (Modules) {
+  function CopyEmbedCode(module) {
+    this.module = module
+    this.copyLink = document.createElement('a')
+    this.copyLink.classList.add('govuk-link')
+    this.copyLink.classList.add('govuk-link__copy-link')
+    this.copyLink.setAttribute('href', '#')
+    this.copyLink.textContent = 'Copy code'
+    this.copyLink.addEventListener('click', this.copyCode.bind(this))
+  }
+
+  CopyEmbedCode.prototype.init = function () {
+    const dd = document.createElement('dd')
+    dd.classList.add('govuk-summary-list__actions')
+    dd.append(this.copyLink)
+
+    this.module.append(dd)
+  }
+
+  CopyEmbedCode.prototype.copyCode = function (e) {
+    e.preventDefault()
+
+    const embedCode = this.module.dataset.embedCode
+    this.writeToClipboard(embedCode).then(this.copySuccess.bind(this))
+  }
+
+  CopyEmbedCode.prototype.copySuccess = function () {
+    const originalText = this.copyLink.textContent
+    this.copyLink.textContent = 'Code copied'
+
+    setTimeout(this.restoreText.bind(this, originalText), 2000)
+  }
+
+  CopyEmbedCode.prototype.restoreText = function (originalText) {
+    this.copyLink.textContent = originalText
+  }
+
+  // This is a fallback for browsers that do not support the async clipboard API
+  CopyEmbedCode.prototype.writeToClipboard = function (embedCode) {
+    return new Promise(function (resolve) {
+      // Create a textarea element with the embed code
+      const textArea = document.createElement('textarea')
+      textArea.value = embedCode
+      // This ensures the textarea doesn't get created at the bottom of the page
```

Comment:
clarification: why do we care where the text area gets created if it's not seen by the user?

Reply:
Good call - I think that was when we were calling `focus()` on it, but we don't need it!